### PR TITLE
provider/kubernetes: Dynamically load namespaces

### PR DESCRIPTION
**Pull request by lwander**
There are two large changes here.

1. The KubernetesProvider now reconfigures its caching agents every 30 seconds
to add new agents for new namespaces
2. Every time namespaces are requested, the clouddriver checks if the namespace
is new, and if so, adds imagePullSecretes for the new namespace

@duftler PTAL

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java [hunk 1]
```diff
@@ -42,43 +46,82 @@ public KubernetesCredentials(KubernetesApiAdaptor apiAdaptor,
...
     this.LOG = LoggerFactory.getLogger(KubernetesCredentials.class);
 
-    for (int i = 0; i < this.dockerRegistries.size(); i++) {
-      LinkedDockerRegistryConfiguration registry = this.dockerRegistries.get(i);
-      this.LOG.info("Adding secrets for docker registry " + registry.getAccountName() + ".");
+    List<String> knownNamespaces = !this.namespaces.isEmpty() ? this.namespaces : apiAdaptor.getNamespacesByName();
+    reconfigureRegistries(knownNamespaces, knownNamespaces);
+  }
+
+  public List<String> getNamespaces() {
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java]: This method could use some comments and probably some unit tests.

**Comment by duftler** [comment]
LGTM except that that one method is a bit hard to follow. Feel free to merge once the comment is addressed.

**Comment by lwander** [comment]
docs & unit tests added
